torch/_export/constraints.py

```diff
@@ -1,9 +1,31 @@
-from typing import Optional
+from typing import Optional, Callable, Union
 
+import torch
+from torch.types import Number, _complex
+from torch import SymInt, SymFloat
 from torch._dynamo import allow_in_graph
-from torch.fx.experimental.symbolic_shapes import constrain_range
+from torch.fx.experimental.symbolic_shapes import constrain_range_non_symint
 from torch.utils._sympy.value_ranges import ValueRangeError
 
+# `Scalar` type used in native_functions.ymal will be translated to `Union[Number, _complex]`
+# could cause type error during since `SymInt` or `SymFloat` will be used.
+# Here manually specify the type explicitly.
+sym_constrain_range: Callable[
+    [Union[Number, _complex, SymInt, SymFloat], Optional[int], Optional[int]],
```

Comment:
Why accept `_complex`? Also, maybe `int, float` instead of `Number`?

Reply:
As mentioned in the comment `Scalar` is translated to python typing as `Union[Number, _complex]` but yeah we could keep it simple and only include `int` and `float` since we're redefining the type anyway.